service/routes.py

```diff
@@ -80,6 +80,33 @@ def get_recommendations(product_id1, product_id2):
         raise NotFound("Recommendation for product id {} and {} was not found.".format(product_id1, product_id2))
     return make_response(jsonify(recommendation.serialize()), status.HTTP_200_OK)
 
+@app.route("/recommendations/products/<int:product_id1>/related-products/<int:product_id2>", methods=["PUT"])
+def update_recommendations(product_id1, product_id2):
+    """
+    update a relationship
+    This endpoint will update a relationship based the data in the body that is posted
+    """
+    app.logger.info("Request to update a ")
+    check_content_type("application/json")
+    recommendation = Recommendation()
+    recommendation.deserialize(request.get_json())
+    old_recommendation = recommendation.find(product_id1, product_id2)
+    if not old_recommendation:
+        raise BadRequest("No existing relationship between product {} and {}".format(product_id1, product_id2))
```

Comment:
Should raise NotFound. No idea how to cover this line.

Reply:
To cover this line you could write a unit test trying to update relationship between -1 and -1 and a NOT FOUND as mentioned above should be expected in your test.

Reply:
done in test_route